From c3e64ec300512231307f684868330ba0a1d06b36 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Thu, 12 May 2011 20:42:52 +0800
Subject: [PATCH 964/998] ENGR00143346 mx53 ard: add touch screen support via I2C2

The capactive touch screen is connected via I2C2 on mx53
ARD RevB board.

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 arch/arm/mach-mx5/mx53_ard.c |   13 ++++++++++++-
 1 files changed, 12 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-mx5/mx53_ard.c b/arch/arm/mach-mx5/mx53_ard.c
index e13430f..c4f3696 100644
--- a/arch/arm/mach-mx5/mx53_ard.c
+++ b/arch/arm/mach-mx5/mx53_ard.c
@@ -502,7 +502,7 @@ static struct mxc_spi_master mxcspi1_data = {
 };
 
 static struct imxi2c_platform_data mxci2c1_data = {
-	.bitrate = 50000,
+	.bitrate = 100000,
 };
 
 static struct imxi2c_platform_data mxci2c2_data = {
@@ -750,6 +750,16 @@ static struct p1003_ts_platform_data p1003_ts_data = {
 
 static struct i2c_board_info mxc_i2c1_board_info[] __initdata = {
 	{
+	 .type = "p1003_fwv33",
+	 .addr = 0x41,
+	 .platform_data = &p1003_ts_data,
+	},
+	{
+	 .type = "egalax_ts",
+	 .addr = 0x4,
+	 .irq  = gpio_to_irq(ARD_TS_INT),
+	},
+	{
 	.type = "cs42888",
 	.addr = 0x48,
 	.platform_data = &cs42888_data,
@@ -1333,6 +1343,7 @@ static void __init mxc_board_init(void)
 	i2c_register_board_info(2, mxc_i2c2_board_info,
 				ARRAY_SIZE(mxc_i2c2_board_info));
 
+
 	mxc_register_device(&mxc_mlb_device, &mlb_data);
 	mx5_set_otghost_vbus_func(mx53_ard_usbotg_driver_vbus);
 	mx5_usb_dr_init();
-- 
1.7.4.1

